regulations/views/preamble.py

```diff
@@ -47,14 +47,15 @@ def get(self, request, *args, **kwargs):
             raise Http404
 
         context = generate_html_tree(subtree)
-        template = context['node']['template_name']
         context['use_comments'] = True
+        template = context['node']['template_name']
 
+        context = {'sub_context': context, 'sub_template': template,
+                   'preamble': preamble}
         if not request.is_ajax() and request.GET.get('partial') != 'true':
-            # Wrap the inner context
-            context = {'sub_context': context, 'sub_template': template,
-                       'preamble': preamble}
             template = 'regulations/preamble-chrome.html'
+        else:
+            template = 'regulations/preamble-partial.html'
```

Comment:
Mmm. Note that you're working against the existing system, here. The conception was that each node would _self_ describe the template which should render it. In this way, we can use that rendered template everywhere (not just when accepting comments, for example).

Is there a reason not to keep the commenting markup within the chrome?